addons/hr_holidays/models/hr.py

```diff
@@ -82,6 +83,70 @@ class Employee(models.Model):
     show_leaves = fields.Boolean('Able to see Remaining Leaves', compute='_compute_show_leaves')
     is_absent_totay = fields.Boolean('Absent Today', compute='_compute_absent_employee', search='_search_absent_employee')
 
+    leave_ids = fields.One2many('hr.leave', 'employee_id', readonly=True)
+    allocation_ids = fields.One2many('hr.leave.allocation', 'employee_id', readonly=True)
+
+    max_leaves = fields.Float(compute='_compute_leaves', string='Maximum Allowed',
+        help='This value is given by the sum of all leaves requests with a positive value.')
+    leaves_taken = fields.Float(compute='_compute_leaves', string='Leaves Already Taken',
+        help='This value is given by the sum of all leaves requests with a negative value.')
+    remaining_leaves = fields.Float(compute='_compute_leaves', string='Remaining Leaves',
+        help='Maximum Leaves Allowed - Leaves Already Taken')
+    virtual_remaining_leaves = fields.Float(compute='_compute_leaves', string='Virtual Remaining Leaves',
+        help='Maximum Leaves Allowed - Leaves Already Taken - Leaves Waiting Approval')
+
+    @api.multi
+    def get_days(self, holiday_types):
```

Comment:
I would rename it to something a bit more explicit. It always bothered me, so while we move it let us rename it :) . Maybe get_leave_days_data ?

Reply:
holiday_type -> leave_type ?

Reply:
I would add a small shortcut: if no holiday_types -> return void structure so that there is no unnecessary query.